genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/grpc/GRpcServicesAutoConfiguration.java

```diff
@@ -63,7 +63,7 @@
     @Bean
     @Lazy
     @ConditionalOnMissingBean(AgentHeartBeatService.class)
-    public AgentHeartBeatService agentHeartBeatService(
+    public GrpcAgentHeartBeatServiceImpl agentHeartBeatService(
```

Comment:
Just curious why changing to `*Impl` instead of the interface? Will this change tie it to a specific implementation?

Reply:
In general for Spring it is a better practice to make the return type as specific as possible. This allows it to be used in the most injection scenarios. If for example someone had a `@ConditionalOnBean(GrpcAgentHeartBeatServiceImpl)` on some other bean definition because they explicitly needed that one even though technically there was a bean of this type available Spring would only know it as `AgentHeartBeatService` and thus that condition would fail.

It doesn't tie anything to a specific implementation just makes the type more concrete in the spring context.